insights/client/utilities.py

```diff
@@ -445,3 +447,41 @@ def os_release_info():
         except Exception as e:
             logger.warning("Failed to detect OS version: %s", e)
     return (os_family, os_release)
+
+
+def largest_spec_in_archive(archive_file):
+    logger.info("Checking for large files...")
+    tar_file = tarfile.open(archive_file, 'r')
+    largest_fsize = 0
+    # get the name of the archive
+    for elems in archive_file.split("/"):
+        if "tar" in elems:
+            name = elems.split(".", 1)
+    name[0]
+
+    # get the archives from inside meta_data directory
+    metadata_top = os.path.join(name[0], "meta_data")
+    data_top = os.path.join(name[0], "data")
+    for file in tar_file.getmembers():
+        if metadata_top in file.name:
+            file_extract = tar_file.extractfile(file.name)
+            specs_metadata = json.load(file_extract)
+            results = specs_metadata.get("results", [])
+            if not results:
+                continue
+            if type(results) != list:
+                results = [results]
+            for result in results:
```

Comment:
It would be better to use `isinstance` builtin instead of type here:

`if isinstance(results, list):`